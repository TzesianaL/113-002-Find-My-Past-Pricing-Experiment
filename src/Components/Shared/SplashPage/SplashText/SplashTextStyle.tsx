import styled from 'styled-components'
/**
 * Styled component is used to decorate SplashText component.
 * Props are passed from ThemeProvider component.
 */
export const SplashTextStyle = styled.div`
  background-color: ${(props) =>
    props.theme.splashScreenText?.backgroundColor ||
    props.theme.backgroundColor ||
    '#6300AA'};
  display: flex;
  flex-direction: column;
  color: ${(props) =>
    props.theme.splashScreenText?.color ||
    props.theme.color ||
    'white'} !important;
  font-size: ${(props) => props.theme.splashScreenText?.fontSize}rem;
  font-family: ${(props) => props.theme.splashScreenText?.fontFamily};
  padding: 16px 1em 24px 1em;
`
