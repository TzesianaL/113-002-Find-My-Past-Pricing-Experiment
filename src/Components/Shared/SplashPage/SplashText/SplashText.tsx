import React, { FC } from 'react'
import Title from 'Components/Shared/Title/Title'
import { SplashTextStyle } from './SplashTextStyle'

export interface SplashTextProps {
  /**
   * Text that displays on the splashscreen component.
   */
  text: string
  /**
   * Title is used to display the title of each text block of splashScreen.
   */
  title?: string
}

/**
 * SplashText component is used to display the text included in the splashScreen
 * SplashTextStyle component is used to style the component
 */
const SplashText: FC<SplashTextProps> = ({ text, title }) => {
  return (
    <SplashTextStyle>
      {title && (
        <Title>
          <div dangerouslySetInnerHTML={{ __html: title }} />
        </Title>
      )}
      <div dangerouslySetInnerHTML={{ __html: text }} />
    </SplashTextStyle>
  )
}

export default SplashText
