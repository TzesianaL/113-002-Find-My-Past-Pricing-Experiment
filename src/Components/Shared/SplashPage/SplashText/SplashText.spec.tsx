import React from 'react'
import 'jest-styled-components'
import * as renderer from 'react-test-renderer'
import SplashText from './SplashText'

describe('<SplashText/>', () => {
  it('should render correctly SplashText', () => {
    const tree = renderer
      .create(<SplashText text=" text title" title="title" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
