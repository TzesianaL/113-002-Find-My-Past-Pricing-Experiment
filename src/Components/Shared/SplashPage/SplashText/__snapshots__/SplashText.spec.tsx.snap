// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplashText/> should render correctly SplashText 1`] = `
.c1 {
  text-align: center;
  text-transform: uppercase;
  font-weight: bold;
  font-family: BoldVf !important;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.2em;
  margin: 0.5em 0em 0.5em 0em;
}

.c0 {
  background-color: #6300AA;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: white !important;
  font-size: rem;
  padding: 16px 1em 24px 1em;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "title",
        }
      }
    />
  </div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": " text title",
      }
    }
  />
</div>
`;
